trio_websocket/__init__.py

```diff
@@ -555,34 +659,72 @@ def port(self):
         constructor), the assigned port will be reflected after calling
         starting the `listen` task.  (Technically, once listen reaches the
         "started" state.)
+
+        This property only works if you have a single listener, and that
+        listener must be socket-based.
         """
-        return self._port
+        if len(self._listeners) > 1:
+            raise RuntimeError('Cannot get port because this server has'
+                ' more than 1 listeners.')
+        try:
+            listener = self.listeners[0]
+            return listener.port
+        except AttributeError:
+            raise RuntimeError('This socket does not have a port: {!r}'
+                .format(listener)) from None
 
-    async def listen(self, *, task_status=trio.TASK_STATUS_IGNORED):
-        ''' Listen for incoming connections. '''
-        if self._ssl is None:
-            serve = partial(trio.serve_tcp, self._handle_connection,
-                self._port, host=self._host,
-                handler_nursery=self._handler_nursery)
-        else:
-            serve = partial(trio.serve_ssl_over_tcp, self._handle_connection,
-                self._port, ssl_context=self._ssl, https_compatible=True,
-                host=self._host, handler_nursery=self._handler_nursery)
+    @property
+    def listeners(self):
+        '''
+        Return a list of listener metadata. Each TCP listener is represented as
+        a ``ListenPort`` instance. Other listener types are represented by their
+        ``repr()``.
+        '''
+        listeners = list()
+        for listener in self._listeners:
+            if isinstance(listener, trio.SocketListener):
+                socket = listener.socket
+                is_ssl = False
+            elif isinstance(listener, trio.ssl.SSLListener):
+                socket = listener.transport_listener.socket
+                is_ssl = True
+            else:
+                socket = None
+            if socket:
+                sockname = socket.getsockname()
+                listeners.append(ListenPort(*sockname, is_ssl))
+            else:
+                listeners.append(repr(listener))
+        return listeners
+
+    async def run(self, *, task_status=trio.TASK_STATUS_IGNORED):
+        '''
+        Start serving incoming connections requests.
```

Comment:
document start vs. start_soon for this function

Reply:
Same as above: I can update documentation for `start()` but not sure what I can do about `start_soon()`.